### DISCUSSION TITLE
console API file output with print

### DISCUSSION THREAD

**Discussion by whittlem** [Textualize/rich#2436] [Q&A]
I would like to use the file output feature of the Console API. When I use the file output, the output to the screen is suppressed. How can I have both? I would like to keep a file copy of everything printed to the screen. The issue I'm having at the moment is I either get the output to the screen or it writes to a file but not both.

Maybe @willmcgugan will know?

**Comment by whittlem** [comment]
I've got it working but maybe it's a hack? I created two consoles. One called console_log which uses the file output option and one standard console called console_term. I just print to both. It works but not sure if it's the best way of doing this.

**Comment by willmcgugan** [comment]
You could create a file-like object, with a write method that writes to both stdout and your file.

**Comment by whittlem** [comment]
Thanks, the way I have it not seems to work.
